railgun/test/integration/controller_test.go

```diff
@@ -72,3 +72,26 @@ func TestProfilingEndpoint(t *testing.T) {
 		return resp.StatusCode == http.StatusOK
 	}, ingressWait, waitTick)
 }
+
+func TestConfigEndpoint(t *testing.T) {
+	if useLegacyKIC() {
+		t.Skip("config endpoint not available in legacy KIC")
+	}
+	assert.Eventually(t, func() bool {
+		successURL := fmt.Sprintf("http://localhost:%v/debug/config/successful", manager.DiagnosticsPort)
+		failURL := fmt.Sprintf("http://localhost:%v/debug/config/failed", manager.DiagnosticsPort)
+		successResp, err := httpc.Get(successURL)
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", successURL, err)
+			return false
+		}
+		defer successResp.Body.Close()
```

Comment:
if you really want the defer work in this function, you need move it above line 84, after line 83

Reply:
Part of 01046f91